pkg/controller/submariner/submariner_controller.go

```diff
@@ -140,6 +144,12 @@ func (r *ReconcileSubmariner) Reconcile(request reconcile.Request) (reconcile.Re
 		}
 	}
 
+	if instance.Spec.ServiceDiscoveryEnabled {
+		if err = r.reconcileServiceDiscoverCR(instance, reqLogger); err != nil {
+			return reconcile.Result{}, err
+		}
+	}
```

Comment:
If `ServiceDiscoveryEnabled` is false then I believe it should delete the ServiceDiscovery resource.

Reply:
Done , I need to add logic to delete the deployed lighthouse-agent in service-discovery controller.